hugolib/site.go

```diff
@@ -1431,27 +1432,47 @@ func (s *Site) assembleMenus() {
 	}
 
 	sectionPagesMenu := s.Info.sectionPagesMenu
-	sectionPagesMenus := make(map[string]interface{})
-	//creating flat hash
 	pages := s.Pages
-	for _, p := range pages {
 
-		if sectionPagesMenu != "" {
+	if sectionPagesMenu != "" {
+		// Create menu entries for section pages with content
+		for _, p := range pages {
+			if p.Section() != "" && p.Kind == KindSection {
```

Comment:
The check for p.Section() was already there before I touched it. You do not want to synthesize a section menu entry for the empty string, and that would happen if there are pages that do not belong to a section.